From 93f11b4a56c47366518979dd8154666bba21d045 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:19 -0400
Subject: [usb] sierra: fix urbs not being killed on shutdown

Message-id: <1410887463-20674-428-git-send-email-dzickus@redhat.com>
Patchwork-id: 95590
O-Subject: [RHEL7 PATCH 427/591] USB: sierra: fix urbs not being killed on shutdown
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit c9d838a898fe232dd51eb924a17cee346bcb697c
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:22:56 2014 +0200

    USB: sierra: fix urbs not being killed on shutdown

    Make sure to stop all I/O, including any active write urbs, at shutdown.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/sierra.c b/drivers/usb/serial/sierra.c
index ed43b18..96ad379 100644
--- a/drivers/usb/serial/sierra.c
+++ b/drivers/usb/serial/sierra.c
@@ -813,6 +813,8 @@ static void sierra_close(struct usb_serial_port *port)
  }
 
  sierra_stop_rx_urbs(port);
+ usb_kill_anchored_urbs(&portdata->active);
+
  for (i = 0; i < portdata->num_in_urbs; i++) {
   sierra_release_urb(portdata->in_urbs[i]);
   portdata->in_urbs[i] = NULL;
-- 
1.7.1